cmd/wallet-sdk-gomobile/credentialquery/credentialquery.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+// Package credentialquery allows querying credentials using presentation definition.
+package credentialquery
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/presexch"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/verifiable"
+	"github.com/piprate/json-gold/ld"
+
+	"github.com/trustbloc/wallet-sdk/cmd/wallet-sdk-gomobile/api"
+	"github.com/trustbloc/wallet-sdk/cmd/wallet-sdk-gomobile/gomobilewrappers"
+	"github.com/trustbloc/wallet-sdk/pkg/credentialquery"
+)
+
+type goAPICredentialQuery interface {
+	Query(query *presexch.PresentationDefinition, contents []*verifiable.Credential) (*verifiable.Presentation, error)
+}
+
+// Query implements querying credentials using presentation definition.
+type Query struct {
+	documentLoader       ld.DocumentLoader
+	goAPICredentialQuery goAPICredentialQuery
+}
+
+// NewQuery returns new Query.
+func NewQuery(documentLoader api.LDDocumentLoader) *Query {
+	wrappedLoader := &gomobilewrappers.DocumentLoaderWrapper{
+		DocumentLoader: documentLoader,
+	}
+
+	return &Query{
+		documentLoader:       wrappedLoader,
+		goAPICredentialQuery: credentialquery.NewInstance(wrappedLoader),
+	}
+}
+
+// Query returns credentials that match PresentationDefinition.
+func (c *Query) Query(query, contents []byte) ([]byte, error) {
```

Comment:
are we adding `credentialReader` option in next PR?

Reply:
Should it be added to only to the core go part(pkg/credentialquery), or also to gomobile bindings?

Reply:
also to go mobile binding.
you can do it in next PR.